components/webvr/webvr_thread.rs

```diff
@@ -251,6 +254,32 @@ impl WebVRThread {
         self.vr_compositor_chan.send(compositor).unwrap();
     }
 
+    fn handle_get_gamepads_for_display(
+        &mut self,
+        display_id: u32,
+        sender: IpcSender<WebVRResult<Vec<(VRGamepadData, VRGamepadState)>>>,
+    ) {
+        match self.service.get_display(display_id) {
+            Some(display) => {
+                let gamepads = display.borrow_mut().fetch_gamepads();
+                match gamepads {
+                    Ok(gamepads) => {
+                        let data = gamepads
+                            .iter()
+                            .map(|g| {
+                                let g = g.borrow();
+                                (g.data(), g.state())
+                            })
+                            .collect();
+                        sender.send(Ok(data)).unwrap();
+                    },
```

Comment:
Nit: can we not use the impl of `collect()` for iterators where `Item=Result<T,E>` to simplify this?

Reply:
Item isn't a result here, the entire list is. We could simplify this a bit with a `.map`, maybe